AntonBabunin/src/main/java/ua/artcode/market/controllers/IAppDbImpl.java

```diff
@@ -0,0 +1,124 @@
+package ua.artcode.market.controllers;
+
+import ua.artcode.market.interfaces.IAppDb;
+import ua.artcode.market.models.Bill;
+import ua.artcode.market.models.Product;
+import ua.artcode.market.models.Salesman;
+import ua.artcode.market.utils.Generator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class IAppDbImpl implements IAppDb {
+
+    private int billNextId;
+    private int productNextId;
+
+    private List<Bill> bills;
+    private Map<Product,Integer> products;
```

Comment:
Map<Product,Integer> you really need Map with integer

Reply:
how else can I store the quantity of goods in the warehouse? "int" I can't use in the Map

Reply:
in method addProduct ITerminalController added reduction in the number of balances in stock when added to a check